lib/greenhouse_io/error.rb

```diff
@@ -1,10 +1,20 @@
+require 'pry'
 module GreenhouseIo
   class Error < StandardError
     attr_reader :code
 
-    def initialize(message, code = nil)
-      super message
+    def initialize(message = nil, code = nil)
+      @message = message
       @code = code
     end
+
+    def message
+      if @message.nil?
+        @code.to_s
```

Comment:
Keeping this backward compatible to those who were checking `error.message == "404"`